### PR TITLE
RAG demo with reranking

### PR DESCRIPTION
**Pull request by dtrawins**
### 🛠 Summary

CVS-153731
RAG demo with full execution delegated to OVMS including reranking

### 🧪 Checklist

- [ ] Unit tests added.
- [x] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

File: demos/continuous_batching/rag/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+```bash
+git clone https://github.com/openvinotoolkit/model_server
+cd model_server/demos/common/export_models
+pip install -q -r requirements.txt
+
+mkdir -p models
+python export_model.py text_generation --source_model meta-llama/Meta-Llama-3-8B-Instruct --weight-format int8 --kv_cache_precision u8 --config_file_path models/config_all.json --model_repository_path models 
+python export_model.py embeddings --source_model Alibaba-NLP/gte-large-en-v1.5 --weight-format int8 --config_file_path models/config_all.json
+python export_model.py rerank --source_model BAAI/bge-reranker-large --weight-format int8  --config_file_path models/config_all.json
+
```
Comment (porlows1) [inline demos/continuous_batching/rag/README.md:15]: Missing ``` (to close bash section)

File: demos/continuous_batching/rag/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+```bash
+git clone https://github.com/openvinotoolkit/model_server
+cd model_server/demos/common/export_models
+pip install -q -r requirements.txt
+
+mkdir -p models
+python export_model.py text_generation --source_model meta-llama/Meta-Llama-3-8B-Instruct --weight-format int8 --kv_cache_precision u8 --config_file_path models/config_all.json --model_repository_path models 
+python export_model.py embeddings --source_model Alibaba-NLP/gte-large-en-v1.5 --weight-format int8 --config_file_path models/config_all.json
+python export_model.py rerank --source_model BAAI/bge-reranker-large --weight-format int8  --config_file_path models/config_all.json
```
Comment (mzegla) [inline demos/continuous_batching/rag/README.md:14]: I'm not that familiar with the final version of the `export_model.py`, but will setting `--config_file_path models/config_all.json` in multiple consecutive executions expand the config? Just making sure there won't be any overwrite.

File: demos/continuous_batching/rag/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+## Deploying the model server
+
+```bash
+docker run -d --rm -p 8000:8000 -v $(pwd)/models:/workspace:ro openvino/model_server:latest --port 9000 --rest_port 8000 --config_path /workspace/config_all.json
```
Comment (mzegla) [inline demos/continuous_batching/rag/README.md:20]: Setting gRPC port looks obsolete
```suggestion
docker run -d --rm -p 8000:8000 -v $(pwd)/models:/workspace:ro openvino/model_server:latest --rest_port 8000 --config_path /workspace/config_all.json
```

File: demos/continuous_batching/rag/rag_demo.ipynb [hunk 4]
```diff
@@ -6,15 +6,15 @@
    "metadata": {},
    "source": [
     "# RAG demo with OpenVINO Model Server and langchain\n",
-    "This demo shows how to use Retrieval Augmented Generation with langchain and OpenAI API.\n",
+    "This demo shows how to use Retrieval Augmented Generation with langchain and gen AI endpoint from OpenVINO Model Server.\n",
```
Comment (mzegla) [inline demos/continuous_batching/rag/rag_demo.ipynb:9]: Gen AI endpoint? Is it not still OpenAI?

File: demos/continuous_batching/rag/rag_demo.ipynb [hunk 5]
```diff
@@ -23,19 +23,9 @@
...
-      "\n",
-      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m24.2\u001b[0m\n",
-      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49m/usr/bin/python3 -m pip install --upgrade pip\u001b[0m\n"
-     ]
-    }
-   ],
+   "outputs": [],
    "source": [
-    "!pip install -q -r requirements.txt"
+    "!pip install  -q -r requirements.txt"
```
Comment (mzegla) [inline demos/continuous_batching/rag/rag_demo.ipynb:28]: Why additional space?

File: demos/continuous_batching/rag/README.md (hunk 2 above)
Comment (dtrawins) [reply demos/continuous_batching/rag/README.md:14]: @mzegla  it will add a model to existing config file

**Review by mzegla — APPROVED** [review]

**Review by porlows1 — APPROVED** [review]

**Review by michalkulakowski — APPROVED** [review]

File: demos/continuous_batching/rag/rag_demo.ipynb [hunk 6]
```diff
@@ -240,7 +231,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "/home/dtrawins/.local/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "/home/dtrawins/model_server/demos/continuous_batching/rag/.venv/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
```
Comment (helena-intel) [inline demos/continuous_batching/rag/rag_demo.ipynb:234]: It would be nice to update ipywidgets so this warning is not shown.

**Review by helena-intel — APPROVED** [review]
I like the central models location! This is a great demo. It worked out of the box for me when I built OVMS from source. If this will be merged before a new OVMS version is released, it would be good to mention the build instructions in the README.

**Pull request merged**